tests/test_metadata_integration.py

```diff
@@ -0,0 +1,348 @@
+
+            # Create model
+            model = exp.create_model(
+                "test_model", run_settings=exp.create_run_settings("echo", ["hello"])
+            )
+
+            # Create ensemble
+            ensemble = exp.create_ensemble(
+                "test_ensemble",
+                run_settings=exp.create_run_settings("echo", ["world"]),
+                replicas=2,
+            )
+
+            # Create database
+            orchestrator = exp.create_database(port=6379, interface="lo")
+
+            # Start all entities - orchestrator and compute entities may create separate run dirs
+            exp.start(orchestrator, block=False)
+            exp.start(model, ensemble, block=False)
+            exp.poll(interval=1)
+
+            # Verify directory structure
+            smartsim_dir = pathlib.Path(temp_dir) / CONFIG.smartsim_base_dir
+            metadata_dir = smartsim_dir / "metadata"
+
+            assert metadata_dir.exists(), "Metadata directory should exist"
+
+            # Check for run-specific subdirectories (may be 1 or 2 depending on timing)
+            run_dirs = [
+                d
+                for d in metadata_dir.iterdir()
+                if d.is_dir() and d.name.startswith("run_")
+            ]
+            assert (
+                len(run_dirs) >= 1
+            ), f"Should have at least one run directory, found: {run_dirs}"
+
+            # Find directory with model/ensemble subdirs
+            run_dir = None
+            for rd in run_dirs:
+                if (rd / "model").exists() or (rd / "ensemble").exists():
+                    run_dir = rd
+                    break
+
+            assert run_dir is not None, "Should find run directory with entity subdirs"
+
+            # Check for entity-specific subdirectories with entity names
+            model_dir = run_dir / "model" / "test_model"
+            ensemble_dir = run_dir / "ensemble" / "test_ensemble"
+
+            assert (
+                model_dir.exists()
+            ), f"Model metadata directory should exist: {model_dir}"
+            assert (
+                ensemble_dir.exists()
+            ), f"Ensemble metadata directory should exist: {ensemble_dir}"
+            # Clean up
+            exp.stop(model, ensemble)
+            exp.stop(orchestrator)
+
+    def test_multiple_experiment_runs_create_separate_run_directories(self):
+        """Test that multiple experiment runs create separate timestamped directories"""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            # First experiment run
+            exp1 = Experiment("test_metadata_run1", exp_path=temp_dir, launcher="local")
+            model1 = exp1.create_model(
+                "test_model1", run_settings=exp1.create_run_settings("echo", ["run1"])
+            )
+
+            exp1.start(model1, block=False)
+            exp1.poll(interval=1)
+            exp1.stop(model1)
```

Comment:
functionally equivalent to
```suggestion
            exp1.start(model1, block=True)
```
? Is this just anticipating a possible future API break?